rust/tools/authorization-logic/src/signing/tink_interface.rs

```diff
@@ -53,6 +53,7 @@ pub fn deserialize_from_file(filename: &str) -> Result<Vec<AstAssertion>, Error>
 // At present, this function is only used in tests, so a warning will be given
 // that this is not used. In the future, a script for generating keys might
 // also use this outside of tests.
+#[allow(dead_code)]
```

Comment:
Code style wise I feel deviant here, in that I just sprinkled this on the one fn that is noted as dead, but perhaps style-wise we should say, if it's an interface and built in a library, we tag all fn's? Or, do it via allow-dead on the BUILD library definition (but that seems wider-swathe and we'd risk missing more).

So I went for minimal intrustion, but feedback is welcome (see issue linked in PR description for more background).

Reply:
I think I like the idea of putting this on all the functions that define interfaces for libraries most. I think doing it this way forces us to look at the public functions and think about if these really should be a part of the library API, or not.

> Or, do it via allow-dead on the BUILD library definition

I'm not sure I follow how this works. Is this a change to the BUILD file rather than the rust code? If so, does this actually stop the cargo warnings?

Reply:
If the function is an interface, it should be `pub` at which point the dead code warning should go away. Rust is pretty good at detecting this kind of thing.